relayer/src/supervisor.rs

```diff
@@ -700,8 +681,54 @@ fn process_batch<Chain: ChainHandle>(
     Ok(())
 }
 
+fn send_telemetry<Src, Dst>(src: &Src, dst: &Dst, events: &[IbcEvent], path: &Packet)
```

Comment:
Can we document this function? Some questions that I think would be helpful for the docs to answer:
1. Do the metrics get sent to both the source and destination? Only one of them? In which situations? 
2. A brief description of each of the metrics that are sent based on the `IbcEvent` variant in the `match`.

Reply:
The idea for these metrics is the following:

* These metrics count the number of `SendPacket`, `WriteAcknowledgement` and `TimeoutPacket` events observed by the supervisor.
* In order to make it easier to see the duality `SendPacket`/`WriteAcknowledgement` or `SendPacket`/`TimeoutPacket`, the `WriteAcknowledgement` and `TimeoutPacket` will have the destination chain labelled as the source as these events are being "sent back" to the chain which submitted the corresponding `SendPacket` event.

So a `SendPacket` event from `ibc-0` will increment the `send_packet_events{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"}` and the returning `WriteAcknowledgment` event will increment the `acknowledgement_events{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer”}`.

But I realise that this might cause confusion. I can update it to label the `chain` and `counterparty` to be the submitting and receiving chains for the isolated events. Meaning the new send/acknowledgement/timeout events metrics label would have:
* `chain` is always the chain that submits the event
* `counterparty` is always the chain that will receive the packet generated from the event

Instead of being the submitting and receiving chain for the whole transfer. Current send/acknowledgement/timeout events metrics label is:
* `chain` being the chain that sends the `SendPacket` event and receives the corresponding `WriteAcknowledgment`, `TimeoutPacket` events
* `counterparty` being the chain that receives the `SendPacket` event and sends the `WriteAcknowledgment`, `TimeoutPacket` events

What do you think?

Reply:
> chain is always the chain that submits the event
counterparty is always the chain that will receive the packet generated from the event

I agree that labeling these in absolute terms, instead of relative terms, is probably a bit easier to parse 🙂